src/MediaPresenter.ts

```diff
@@ -74,52 +73,53 @@ export class MediaPresenter {
     if (this.media.length === 0) {
       return "";
     }
-    return this.media[fileIndex % this.filesLength].file.name;
+    return this.getMedia(fileIndex).file.name;
   }
 
   getFile(fileIndex: number): File {
-    return this.media[fileIndex % this.filesLength].file;
+    return this.getMedia(fileIndex).file;
   }
 
   @mobx.computed
   get filesLength(): number {
     return this.media.length;
   }
 
+  // Check if any file is uploaded
+  @mobx.computed
+  mediaReady() {
+    return this.filesLength !== 0;
+  }
+
   // shows how long each media has played
+  // mod by duration, to loop back to the beginning of media
   @mobx.action
   incrementFilePlayed(fileIndex: number, seconds: number): void {
-    let index = fileIndex % this.filesLength;
-    this.media[index].played =
-      (this.media[index].played + seconds) % this.media[index].duration;
+    let { duration, played } = this.getMedia(fileIndex);
+    played = (played + seconds) % duration;
   }
 
   // reset all played media to 0
   @mobx.action
   resetAllPlayedFiles(): void {
-    for (let index = 0; index < this.filesLength; index++) {
-      this.media[index].played = 0;
+    for (let fileIndex = 0; fileIndex < this.filesLength; fileIndex++) {
+      this.getMedia(fileIndex).played = 0;
     }
   }
 
   getFilePlayed(fileIndex: number): number {
-    if (
-      fileIndex % this.filesLength === 0 ||
-      this.getDuration(fileIndex % this.filesLength) === 0
-    )
+    // making sure that we don't meet the case of 0 % 0 which is NaN
+    let { duration, played } = this.getMedia(fileIndex);
+    let computedIndex = fileIndex % this.filesLength;
+    if (!computedIndex || !this.getDuration(computedIndex)) {
```

Comment:
`this.getDuration(computedIndex)` (and below) is the same as the `duration` you've destructured above